File: 프로그래머스/1/12926. 시저 암호/시저 암호.cs
using System.Text;
public class Solution {
    public string solution(string s, int n) {
        StringBuilder answer = new StringBuilder();
        for(int i = 0; i < s.Length; i++)
        {
            char c = s[i];
            if(c == ' ') answer.Append(' ');
            else if(char.IsUpper(c))
            {
                char shifted = (char)((c - 'A' + n) % 26 + 'A');
                answer.Append(shifted);
            }
            else if(char.IsLower(c))
            {
                char shifted = (char)((c - 'a' + n) % 26 + 'a');
                answer.Append(shifted);
            }
        }
        return answer.ToString();
    }
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    for(char &c : s)
    {
        if(c >= 'a' && c <= 'z')
        {
            c = (c - 'a' + n) % 26 + 'a';
        }
        else if(c >= 'A' && c <= 'Z')
            c = (c - 'A' + n) % 26 + 'A';
    }
        
    return s;
}